crates/runtime/src/embeddings/index/scan_table.rs

```diff
@@ -14,136 +14,143 @@ See the License for the specific language governing permissions and
         filters: &[Expr],
-    ) -> DataFusionResult<TableScan> {
-        let num_underlying_columns = self.table_provider.schema().fields().len();
-        let underlying_projection = projection.map(|proj| {
-            proj.iter()
-                .filter(|&idx| *idx < num_underlying_columns)
-                .copied()
-                .collect()
-        });
-
-        let filter_refs: Vec<&Expr> = filters.iter().collect();
-        let underlying_filters = self
-            .table_provider
-            .supports_filters_pushdown(filter_refs.as_slice())?
-            .into_iter()
-            .zip(filters.iter())
-            .filter_map(|(supported, filter)| {
-                if matches!(supported, TableProviderFilterPushDown::Unsupported) {
-                    None
-                } else {
-                    Some(filter.clone())
-                }
-            })
-            .collect::<Vec<_>>();
-
-        TableScan::try_new(
-            TableReference::parse_str("base_table"),
-            Arc::new(DefaultTableSource::new(Arc::clone(&self.table_provider))),
-            underlying_projection,
-            underlying_filters,
-            None,
-        )
+    ) -> bool {
+        if !projection.is_subset(
+            &schema
+                .iter()
+                .map(|f| f.name().to_string())
+                .collect::<HashSet<String>>(),
+        ) {
+            // schema does not have all columns.
+            return false;
+        }
+        // Ensure filters do not reference column not in the schema
+        columns_missing_from(filters, schema).is_empty()
     }
 
-    /// For a projection relative to [`VectorScanTableProvider`], check if the embedding column is being requested.
-    fn need_vector_column(&self, projection: Option<&Vec<usize>>) -> bool {
-        let Some(proj) = projection else {
-            return true; // None projection -> "SELECT *".
+    async fn apply_proj_and_filter(
+        &self,
+        state: &dyn Session,
+        lp: Arc<LogicalPlan>,
+        projection: &HashSet<String>,
+        filters: &[Expr],
+    ) -> Result<Arc<dyn ExecutionPlan>, DataFusionError> {
+        let filtered = if let Some(filter) = filters.iter().cloned().reduce(Expr::and) {
+            Arc::new(LogicalPlan::Filter(Filter::try_new(filter, lp.into())?))
+        } else {
+            lp
         };
 
-        let Some(idx) = index_of_column(
-            &self.schema(),
-            embedding_col!(self.index.search_column()).as_str(),
-        ) else {
-            return false; // Technically unreachable, but by definition not needed.
-        };
+        let lp = LogicalPlan::Projection(Projection::try_new(
+            projection
+                .iter()
+                .sorted_unstable()
+                .map(|p| Expr::Column(Column::new_unqualified(p.clone())))
+                .collect(),
+            filtered,
+        )?);
 
-        proj.contains(&idx)
+        state.create_physical_plan(&lp).await
```

Comment:
Is there a reason we need to create the physical plan here instead of just returning a LogicalPlan and doing the physical plan creation once?